src/app/[locale]/posts/(components)/PostsPage.tsx

```diff
@@ -311,7 +367,13 @@ function CustomHits({ categories }: Pick<Props, "categories">) {
       >
         {hits.map((hit, i) => {
           // todo: add a featured image once we have image templates in place
-
+          const fullText = concatenateBodies(hit.blocks);
+          let timeToRead;
+          if (hit.post_type === "video") {
+            timeToRead = `${formatDuration(hit.video?.data?.contentDetails?.duration || '')} watch`;
+          } else {
+            timeToRead = `${calculateReadingTime(fullText)} read`;
```

Comment:
too much calculations on each iteration move to `cms-scripts`

Reply:
Thanks for the comment, Tsotne, I moved it to fileToPost function.